src/types.rs

```diff
@@ -703,6 +729,19 @@ impl fmt::Display for CertificateIdentifier<'_> {
     }
 }
 
+fn deserialize_static_certificate_identifier<'de, D>(
+    deserializer: D,
+) -> Result<Option<CertificateIdentifier<'static>>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let opt_cert_id = Option::<CertificateIdentifier<'_>>::deserialize(deserializer)?;
+    Ok(opt_cert_id.map(move |cert_id| CertificateIdentifier {
+        authority_key_identifier: Cow::Owned(cert_id.authority_key_identifier.into_owned()),
+        serial: Cow::Owned(cert_id.serial.into_owned()),
+    }))
```

Comment:
Maybe reformulate more like this?

```rust
let Some(cert_id) = Option::<CertificateIdentifier<'_>>::deserialize(deserializer)? else {
    return Ok(None);
};

Ok(CertificateIdentifier { .. })
```

Also I'd suggest outlining the `into_owned()` into a `CertificateIdentifier::into_owned()` so as to make this code accessible to outside callers as well.

Reply:
Done x2. I put the `into_owned()` addition in its own commit up-front.